pkg/server/settings_cache.go

```diff
@@ -99,6 +99,28 @@ var _ settingswatcher.Storage = (*settingsCacheWriter)(nil)
 func storeCachedSettingsKVs(ctx context.Context, eng storage.Engine, kvs []roachpb.KeyValue) error {
 	batch := eng.NewBatch()
 	defer batch.Close()
+
+	// Read all existing cached settings keys
+	existingKVs, err := storage.MVCCScan(ctx, eng, keys.LocalMax, roachpb.KeyMax, hlc.MaxTimestamp, storage.MVCCScanOptions{})
+	if err != nil {
+		return err
+	}
```

Comment:
I believe the bounds of this scan should be `keys.LocalStoreCachedSettingsKeyMin` to `keys.LocalStoreCachedSettingsKeyMax` unless I'm misunderstanding?